src/BalanceClient.ts

```diff
@@ -0,0 +1,60 @@
+const {providers} = require("near-api-js");
+import {Network} from "./Utils";
+import {getRpcUrl} from "./NearConfig";
+
+export class BalanceClient {
+
+  static async ftBalanceOfOnNear(network: Network, contractId: string, accountId: string): Promise<number> {
+    const connectionInfo = { url: getRpcUrl(network) };
+    const provider = new providers.JsonRpcProvider(connectionInfo);
+
+    const args = `{ "account_id": "${accountId}" }`;
+    const encodedArgs = Buffer.from(args).toString("base64");
+
+    const queryResponse = await provider.query({
+      request_type: "call_function",
+      account_id: contractId,
+      method_name: "ft_balance_of",
+      args_base64: encodedArgs,
+      finality: "final",
+    });
+
+    return JSON.parse(Buffer.from(queryResponse.result).toString());
```

Comment:
all methods in this file can be even more extracted, given that queryResponse up until return are always same, except for method name. I would propose extracting that.

Reply:
Good point, I extracted this into a separate utils function `callViewMethod` and I used it for those 3 balance client functions and in light client function. Also, the `args` part is different in balance client functions so that is the argument to the shared method.